modules/websearch/lib/websearch_webinterface.py

```diff
@@ -159,6 +167,48 @@ def wash_search_urlargd(form):
 
     return argd
 
+
+def get_year(recid):
+    for tag in ["773__y", "260__c", "269__c", "909C4y", "925__a"]:
+        date = get_fieldvalues([recid], tag)
+
+        if len(date) == 1:
+            date = date[0]
+            match_obj = re.search("\d\d\d\d", date)
+            if match_obj is not None:
+                return int(match_obj.group())
```

Comment:
@glouppe you can better retrieve the year using `bibrec.earliest_date` which is maintained by a bibcheck scripts that looks into these and many more fields.